streambed/src/crypto.rs

```diff
@@ -1,35 +1,41 @@
 //! Helpers to encrypt and decrypt things, particularly message payloads.
 
 use aes::{
-    cipher::{generic_array::GenericArray, BlockEncrypt, KeyInit},
+    cipher::{generic_array::GenericArray, BlockEncrypt},
     Aes128,
 };
 use rand::RngCore;
+use sha2::{Digest, Sha256, Sha512};
+
+use hmac::Hmac;
+type HmacSha256 = Hmac<Sha256>;
 
 const BLOCKSIZE: usize = 16;
 
+pub const KEY_SIZE: usize = 16;
 pub const SALT_SIZE: usize = 12;
 
 /// Generate a random salt for the required SALT_SIZE.
-pub fn salt<T>(rng: &mut T) -> Vec<u8>
+pub fn salt<T>(rng: &mut T) -> [u8; SALT_SIZE]
 where
     T: RngCore,
 {
-    let mut bytes = vec![0; SALT_SIZE];
+    let mut bytes = [0; SALT_SIZE];
     rng.fill_bytes(&mut bytes);
     bytes
 }
 
 /// Performs an AES-128 ECB decryption using a salt and a 16 byte key
 #[inline]
-pub fn decrypt(bytes: &mut [u8], key: &[u8], salt: &[u8]) {
+pub fn decrypt(bytes: &mut [u8], key: &[u8; KEY_SIZE], salt: &[u8; SALT_SIZE]) {
     encrypt(bytes, key, salt)
 }
```

Comment:
`decrypt` calls `encrypt` here - this seems odd

Reply:
It’s the way aes encryption works. Different function names convey intent.

Reply:
Ah, interesting, maybe a comment to explain that would be good :)